core/src/main/java/tc/oc/pgm/flag/state/Uncarried.java

```diff
@@ -72,21 +74,11 @@ protected void placeBanner() {
       PGM.get().getGameLogger().severe("Failed to place banner at " + this.location);
     }
 
-    this.labelEntity =
-        this.location.getWorld().spawn(this.location.clone().add(0, 1.7, 0), ArmorStand.class);
```

Comment:
you're loosing this effectively-speaking location-set, that's relevant because the flag post moves, but your hologram does not

Reply:
Digging in code, it seems like the state is created at the time the transition is queued. So it would use the location at the time of transition-start anyways. The only thing happening between constructor and `#enterState` being called is leaving the old state